src/clients/openai.rs

```diff
@@ -476,6 +580,22 @@ pub struct ChatCompletionChunkChoice {
     pub index: u32,
 }
 
+/// A chat completion delta generated by streamed model responses.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ChatCompletionDelta {
+    /// The contents of the message.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub content: Option<String>,
+    /// The refusal message generated by the model.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub refusal: Option<String>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub tool_calls: Vec<ToolCall>,
+    /// The role of the author of this message.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub role: Option<String>,
```

Comment:
I made it `Option` for the `ChatCompletionDelta` object (used in streaming responses) since their documentation has examples without it included, e.g.  2 below. However, I'm not aware of a scenario where it wouldn't be. Since this isn't created by the user, it doesn't really matter though.

```
{"id":"chatcmpl-123","object":"chat.completion.chunk","created":1694268190,"model":"gpt-4o-mini", "system_fingerprint": "fp_44709d6fcb", "choices":[{"index":0,"delta":{"role":"assistant","content":""},"logprobs":null,"finish_reason":null}]}

{"id":"chatcmpl-123","object":"chat.completion.chunk","created":1694268190,"model":"gpt-4o-mini", "system_fingerprint": "fp_44709d6fcb", "choices":[{"index":0,"delta":{"content":"Hello"},"logprobs":null,"finish_reason":null}]}
```